src/llm/legacy/servable.cpp

```diff
@@ -0,0 +1,158 @@
+#pragma GCC diagnostic pop
+#pragma warning(pop)
+
+#include "../../http_payload.hpp"
+#include "../../mediapipe_internal/mediapipe_utils.hpp"
+#include "../apis/openai_completions.hpp"
+#include "../text_processor.hpp"
+#include "servable.hpp"
+
+namespace ovms {
+
+static std::string wrapTextInServerSideEventMessage(const std::string& text) {
+    std::stringstream ss;
+    ss << "data: " << text << "\n\n";
+    return ss.str();
+}
+
+// Node resources interface start
+std::shared_ptr<GenAiServableExecutionContext> LegacyServable::createExecutionContext() {
+    return std::make_shared<LegacyServableExecutionContext>();
+}
+
+std::shared_ptr<GenAiServableProperties> LegacyServable::getProperties() {
+    return properties;
+}
+
+absl::Status LegacyServable::scheduleExecution(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    properties->legacyExecutor->addRequest(legacyExecutionContext);
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::readCompleteExecutionResults(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    auto legacyExecutionContext = std::static_pointer_cast<LegacyServableExecutionContext>(executionContext);
+    if (legacyExecutionContext->payload.client->isDisconnected()) {
+        return absl::CancelledError();
+    }
+    legacyExecutionContext->finished.wait();
+    // std::lock_guard<std::mutex> guard(pipelineMutex);
+    // legacyExecutionContext->results = properties->pipeline->generate(legacyExecutionContext->inputIds, legacyExecutionContext->apiHandler->createGenerationConfig());
+    return absl::OkStatus();
+}
+
+absl::Status LegacyServable::parseRequest(std::shared_ptr<GenAiServableExecutionContext>& executionContext) {
+    executionContext->apiHandler = std::make_shared<OpenAIChatCompletionsHandler>(*executionContext->payload.parsedJson,
+        executionContext->endpoint,
+        std::chrono::system_clock::now(),
+        getProperties()->tokenizer);
+
+    auto status = executionContext->apiHandler->parseRequest(getProperties()->maxTokensLimit, getProperties()->bestOfLimit, getProperties()->isSpeculativePipeline);
+    if (!status.ok()) {
+        SPDLOG_LOGGER_ERROR(llm_calculator_logger, "Failed to parse request: {}", status.message());
+        return status;
+    }
+
+    if (executionContext->apiHandler->isStream()) {
+        executionContext->lastStreamerCallbackOutput = "";  // initialize with empty string
+        auto callback = [& lastStreamerCallbackOutput = executionContext->lastStreamerCallbackOutput](std::string text) {
+            SPDLOG_LOGGER_TRACE(llm_calculator_logger, "Streamer callback executed with text: [{}]", text);
+            lastStreamerCallbackOutput += text;
```

Comment:
Could you check if the same approach works for CB pipelines? 
In both cases we clear `lastStreamerCallbackOutput` after we read from it, so in token-by-token flow of CB pipeline it should not make a difference whever we assign or append. If it works, please add comments so it is clear why we do it that way and why we can do it.